sdk/storage/Azure.Storage.Blobs/src/BlobExtensions.cs

```diff
@@ -764,6 +764,7 @@ internal static BlobProperties ToBlobProperties(this ResponseWithHeaders<BlobGet
                 ArchiveStatus = response.Headers.ArchiveStatus,
                 AccessTierChangedOn = response.Headers.AccessTierChangeTime.GetValueOrDefault(),
                 VersionId = response.Headers.VersionId,
+                IsLatestVersion = response.Headers.IsCurrentVersion.GetValueOrDefault(),
```

Comment:
another data point to support this https://github.com/Azure/azure-sdk-for-net/pull/21032#discussion_r644902513 .
@amnguye could you check what would it take to convert internal sets to internal ctor for models we care most about?
If this sounds reasonable we should create issue/work item to get this addressed.

Reply:
It wouldn't be too bad, there's only like 6 places I really need to change. I can do it for BlobProperties in this PR.

I made an issue for the rest of the models here: https://github.com/Azure/azure-sdk-for-net/issues/21553